### ISSUE TITLE
Mocha detects global variable leak of "token"

### ISSUE THREAD

**Issue by kasbah** [Hardmath123/nearley#289]
Hey, thanks very much for nearly. I built a [parser for electronic component descriptions](https://github.com/monostable/electro-grammar) using it and it works really well ([demo](https://monostable.github.io/electro-grammar/)).

I noticed that when I run my mocha tests with global variable leak detection it detects a global of `token`.

```
$ mocha --check-leaks
...
Error: global leak detected: token
```

I assume this is a variable used while running the parser but haven't been able to track it down. Is there a good reason for having it global? 

EDIT: Just noticed that my project is, already linked in the docs! I must have mentioned it in a previous issue. Anyway, I have renamed it from `npm-parse-electronic-component` to `electro-grammar`.

**Comment by tjvr** [comment]
I didn't know about `--check-leaks`, thanks! I squashed a couple of global leaks in 6467265.

I can't find `token =` anywhere, though. Are you sure it's not in your code? :-)

> I must have mentioned it in a previous issue

Not necessarily—@Hardmath123 is pretty good about spotting people using nearley in the wild!

**Comment by Hardmath123** [comment]
Glad you liked nearley, @kasbah! I updated the website with the new name — I really like your project. "Lightning fast parser" is a great description for a project whose logo is a bolt! ;-)

(By the way, wrt the demo page, the project name is "nearl[e]y", not "nearly".)

**Comment by kasbah** [comment]
Thanks! To be honest it's not really a logo, I just put lightning bolts everywhere because it's in all the emoji and icon sets. I keep making that mistake with the name nearly everywhere. Will correct it. In regards to the global leak issue, looks like I was using an older version and this was fixed with 60c7d79fe21290de4!

**Comment by bd82** [comment]
> I can't find token = anywhere, though. Are you sure it's not in your code? :-)

You may be able to use a **setter** to find out who is setting the global token.
```javascript
var myGlobal = ... // window (browser) OR global (node.js)
Object.defineProperty(myGlobal, 'token', { set: function(x) { debugger; } });
```
